fe/fe-core/src/main/java/com/starrocks/catalog/MaterializedView.java

```diff
@@ -485,6 +488,10 @@ public Set<String> getUpdatedPartitionNamesOfTable(Table base) {
                 result.add(basePartitionName);
             }
         }
+        if (withMv && baseTable.isMaterializedView()) {
+            Set<String> partitionNames = ((MaterializedView) baseTable).getPartitionNamesToRefreshForMv();
```

Comment:
Why check is isMaterializedView in this function instead of check this outside and call getPartitionNamesToRefreshForMv dirctly?

Reply:
because here we should consider the changed partitions of base tables, not the mv above the base tables